### DISCUSSION TITLE
Any rule that covers this?

### DISCUSSION THREAD

**Discussion by EloB** [eslint/eslint#16551] [Q&A]
I'm looking for something like a `no-lonely-if` in functions to prevent unnecessary blocks. Would be really good if it had autofix as well to be able to opposite the condition.

One less line and less nesting. Seems like a winner for me. Especially if you come to codebase that has a lot of this.

## Bad
```js
function flipACoin() {
  if (Math.random() > 0.5) {
    console.log("Hit");
  }
}
```

## Good
```js
function flipACoin() {
  if (Math.random() <= 0.5) return;
  console.log("Hit");
}
```

**Comment by mdjermanovic** [comment]
There is no core rule for this. I think the `@shopify/prefer-early-return` plugin rule might be what you're looking for:

https://github.com/Shopify/web-configs/blob/main/packages/eslint-plugin/docs/rules/prefer-early-return.md

It doesn't seem autofixable though.

**Reply by EloB** [reply]
Thanks! I maybe can add auto fixable.
